nixos/modules/services/networking/networkmanager.nix

```diff
@@ -492,6 +543,30 @@ in {
       aliases = [ "dbus-org.freedesktop.nm-dispatcher.service" ];
     };
 
+    systemd.services.NetworkManager-ensureProfiles = mkIf (cfg.profiles != { }) {
+      description = "This unit creates network-manager profiles";
+      wantedBy = [ "multi-user.target" ];
+      before = [ "network-online.target" ];
+      script = let
+        path = id: "/run/NetworkManager/system-connections/${id}.nmconnection";
+      in ''
+        mkdir -p /run/NetworkManager/system-connections
+      '' + lib.concatMapStringsSep "\n"
+        (profile: ''
+          ${pkgs.envsubst}/bin/envsubst -i ${ini.generate (lib.escapeShellArg profile.n) profile.v} > ${path (lib.escapeShellArg profile.n)}
```

Comment:
What do you mean with deletion? If it's not in the config anymore it will be gone after the next boot because they are saved in the temp directory of NM which gets cleared every boot otherwise they will just be overridden. I guess I should add this to the module description.

Reply:
Yes, exactly, this is what I meant and I think it's good to mention it :-)